jerry-core/vm/vm.c

```diff
@@ -794,92 +769,63 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
             break;
           }
         }
-        if (CBC_BRANCH_IS_BACKWARD (opcode_flags))
+
+        if (opcode_data & VM_OC_BACKWARD_BRANCH)
         {
           branch_offset = -branch_offset;
         }
       }
 
-      free_flags = 0;
+      left_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+      right_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+
       if (VM_OC_HAS_GET_ARGS (opcode_data))
       {
-        uint32_t operands = VM_OC_GET_ARGS_GET_INDEX (opcode_data);
+        uint32_t operands = VM_OC_GET_ARGS_INDEX (opcode_data);
 
-        if (operands >= VM_OC_GET_ARGS_GET_INDEX (VM_OC_GET_LITERAL))
+        if (operands >= VM_OC_GET_LITERAL)
         {
           uint16_t literal_index;
           READ_LITERAL_INDEX (literal_index);
           READ_LITERAL (literal_index,
-                        left_value,
-                        free_flags = VM_FREE_LEFT_VALUE);
+                        left_value);
 
-          switch (operands)
+          if (operands != VM_OC_GET_LITERAL)
```

Comment:
Because all the predicates are equality checks on the same variable, `operands`. That's the typical use case for a switch.
The braces around the cases are no extras, they are already there for the if/if-else/else branches. (To be fair: some extra comes from the breaks and the case for `VM_OC_GET_LITERAL`.)

Anyway, I just wanted to point out that nesting `if`s with equality checks on the same var is not that nice. But also mentioned that it's a matter of taste.